chapters/api_concepts.txt

```diff
@@ -450,12 +450,12 @@ configuration (size, dimensionality, etc.) is discarded in favor of the
 currently mapped array configuration.
 
 Devices are only required to allocate directly mapped arrays for parameters
-corresponding extensions that the device implements. In cases where the
+corresponding to extensions that the device implements. In cases where the
 parameter is unknown by the device, the device is permitted to return `NULL`.
 
 Devices are permitted to have a non-dense element stride, which is returned by
-writing to `elementStride`. Implementations are permitted to return `0` in the
```

Comment:
If the device is "permitted" it should strictly return 0 for dense arrays. The API user shouldn't have to deal with ambiguity of both a `0` stride and a `sizeof(element)` redundantly expressing dense packing.